aws/sdk/integration-tests/s3/tests/throughput-timeout.rs

```diff
@@ -0,0 +1,68 @@
+use aws_smithy_async::future::fn_stream::FnStream;
+use aws_smithy_async::rt::sleep::AsyncSleep;
+use aws_smithy_http::body::SdkBody;
+use aws_smithy_http::byte_stream::ByteStream;
+use aws_smithy_http::minimum_throughput::MinimumThroughputBody;
+use std::convert::Infallible;
+use std::time::Duration;
+
+// This test is flaky. The error message will end with something like "0.999 B/s was observed"
+#[should_panic = "minimum throughput was specified at 2 B/s, but throughput of"]
+#[tokio::test]
+async fn test_throughput_timeout_happens_for_slow_stream() {
+    tracing_subscriber::fmt::init();
+    tracing::info!("tracing is working");
+    // let test_start_time = std::time::Instant::now();
+    // Have to return results b/c `hyper::body::Body::wrap_stream` expects them
+    let stream: FnStream<Result<String, Infallible>, _> = FnStream::new(|tx| {
+        let async_sleep = aws_smithy_async::rt::sleep::TokioSleep::new();
+        Box::pin(async move {
+            for i in 0..10 {
+                // Will send slightly less that 1 byte per second because ASCII digits have a size
+                // of 1 byte and we sleep for 1 second after every digit we send.
+                tx.send(Ok(format!("{}", i))).await.expect("failed to send");
+                async_sleep.sleep(Duration::from_secs(1)).await;
```

Comment:
Ideally, we should be able to fake time keeping in the `MinimumThroughputBody` so that we don't need to sleep in tests. Essentially, be able to replace the source of `Instant::now()` inside of `MinimumThroughputBody`.